server/data/data-api/src/test/java/org/apache/james/rrt/lib/CanSendFromContract.java

```diff
@@ -76,6 +75,17 @@ default void userCanSendFromShouldBeTrueWhenSenderIsAnAliasOfAnAliasOfTheUser()
         assertThat(canSendFrom().userCanSendFrom(USER, userAliasBis)).isTrue();
     }
 
+    @Test
+    default void userCannotSendFromShouldBeTrueWhenSenderIsAnAliasOfAnAliasOfAnAliasOfTheUser() throws Exception {
```

Comment:
s/Cannot/Can

Reply:
Not with a two-level recursion

Reply:
then it should be `userCanSendFromShouldBeFalse...`
but I read:
```
assertThat(canSendFrom().userCanSendFrom(USER, userAliasBis)).isTrue();
```
in your test.
So you are missing:
```
assertThat(canSendFrom().userCanSendFrom(USER, userAliasTer)).isFalse();
```